Cleanup webpushd.plist install location

beidson opened the pull request:
#### a7616f1b230beece9e6daef6432e68526fa90fa5
<pre>
Cleanup webpushd.plist install location
<a href="https://bugs.webkit.org/show_bug.cgi?id=263127">https://bugs.webkit.org/show_bug.cgi?id=263127</a>
rdar://116030136

Reviewed by Elliott Williams.

Restore the SYSTEM_SECONDARY_CONTENT_PATH behavior lost in a previous patch.

* Source/WebKit/Configurations/webpushd.xcconfig:

Canonical link: <a href="https://commits.webkit.org/269331@main">https://commits.webkit.org/269331@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fb98cc0821e2c4ba0be3148d068a3a2a40ad4c13

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22230 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/22433 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23302 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24126 "Failed to checkout and rebase branch from PR 19056") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20576 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/26692 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22757 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/24126 "Failed to checkout and rebase branch from PR 19056") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22463 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/22173 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19286 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24980 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/19225 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26391 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20240 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20371 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24256 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20807 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17703 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/20162 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5293 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/24372 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20754 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/Configurations/webpushd.xcconfig:
```diff
@@ -66,8 +66,9 @@ LAUNCHD_PLIST_INSTALL_PATH = $(LAUNCHD_PLIST_INSTALL_PATH_$(WK_RELOCATABLE_WEBPU
-LAUNCHD_PLIST_INSTALL_PATH_NO[sdk=macosx*] = /System/Library/LaunchAgents
+LAUNCHD_PLIST_INSTALL_PATH_NO = $(LAUNCHD_PLIST_INSTALL_PATH_NO_$(USE_SYSTEM_CONTENT_PATH))
+LAUNCHD_PLIST_INSTALL_PATH_NO_YES = /System/Library/LaunchDaemons
+LAUNCHD_PLIST_INSTALL_PATH_NO_YES[sdk=macos*] = $(SYSTEM_SECONDARY_CONTENT_PATH)/System/Library/LaunchAgents
```
Seems like we are assuming USE_SYSTEM_CONTENT_PATH is always YES. Should be fine, since downlevel updates have WK_RELOCATABLE_WEBPUSHD, so they will use the path above instead of expanding this setting to the empty string.

But in that case, I think we can just drop the USE_SYSTEM_CONTENT_PATH check altogether:
```suggestion
LAUNCHD_PLIST_INSTALL_PATH_NO = /System/Library/LaunchDaemons
LAUNCHD_PLIST_INSTALL_PATH_NO[sdk=macos*] = $(SYSTEM_SECONDARY_CONTENT_PATH)/System/Library/LaunchAgents
```

beidson committed:
Cleanup webpushd.plist install location
https://bugs.webkit.org/show_bug.cgi?id=263127
rdar://116030136

Reviewed by Elliott Williams.

Restore the SYSTEM_SECONDARY_CONTENT_PATH behavior lost in a previous patch.

* Source/WebKit/Configurations/webpushd.xcconfig:

Canonical link: https://commits.webkit.org/269331@main
